Packages/com.chisel.core/Chisel/Core/GeneratorBase/BrushFactory.Utility.cs

```diff
@@ -1233,5 +1233,47 @@ public static void CreateExtrudedSubMesh(ref BrushMesh brushMesh, Vector3[] side
             if (!brushMesh.Validate(logErrors: true))
                 brushMesh.Clear();
         }
+
+        /// <summary>
+        /// Creates a brush out of a set of planes. A brush consists of several faces, each of which can be defined by a plane (of which there must be at least four).
+        /// Each plane defines a half space, that is, an infinite set of points that is bounded by a plane. The intersection of these half spaces forms a convex polyhedron.
+        /// The brush geometry is not centered unless your planes are, use <seealso cref="BrushMesh.CenterAndSnapPlanes"/> to center and position the game object to match.
+        /// <para>Note: The current algorithm only works within a limited range of +-4096 world units from the world center. If possible, perform your operations near the center of the world for optimal accuracy.</para>
+        /// </summary>
+        public static void CreateFromPlanes(ref BrushMesh brushMesh, float4[] planes, in ChiselSurfaceDefinition surfaceDefinition)
+        {
+            Debug.Assert(planes != null && planes.Length >= 4);
+
+            // TODO: check if we can cut up a box with infinite dimensions to get around the size constraint.
+
+            // build a very large cube brush.
+            surfaceDefinition.EnsureSize(6);
+            CreateBox(ref brushMesh, new Vector3(-4096, -4096, -4096), new Vector3(4096, 4096, 4096), in surfaceDefinition);
```

Comment:
until we find a good way to find the bounds of the plane intersections, if we have a version of this method that takes a bounds, then we can just use the bounds of all the brushes that are being merged

Reply:
I have added a bounds parameter. This completely replaced the hard-coded range limit and improved the accuracy.